src/api/mod.rs

```diff
@@ -1819,47 +1818,55 @@ impl<T: Pixel> ContextInner<T> {
           },
           TxSize::TX_8X8,
           fi.sequence.bit_depth,
-          Some(PredictionMode::DC_PRED),
+          None,
         );
 
-        let mut plane_after_prediction_region = plane_after_prediction
-          .region_mut(Area::Rect {
-            x: (x * IMPORTANCE_BLOCK_SIZE) as isize,
-            y: (y * IMPORTANCE_BLOCK_SIZE) as isize,
-            width: IMPORTANCE_BLOCK_SIZE,
-            height: IMPORTANCE_BLOCK_SIZE,
-          });
-
-        PredictionMode::DC_PRED.predict_intra(
-          TileRect {
-            x: x * IMPORTANCE_BLOCK_SIZE,
-            y: y * IMPORTANCE_BLOCK_SIZE,
-            width: IMPORTANCE_BLOCK_SIZE,
-            height: IMPORTANCE_BLOCK_SIZE,
-          },
-          &mut plane_after_prediction_region,
-          TxSize::TX_8X8,
-          fi.sequence.bit_depth,
-          &[], // Not used by DC_PRED.
-          0,   // Not used by DC_PRED.
-          &edge_buf,
-        );
+        let bit_depth = self.config.bit_depth;
+        let intra_mode_set = RAV1E_INTRA_MODES;
+        let intra_cost = intra_mode_set
+          .iter()
+          .map(|&luma_mode| {
+            let mut plane_after_prediction_region = plane_after_prediction
```

Comment:
The tile and the region seem to stay the same, is there a reason why they are in the iteration instead of out?

Reply:
Borrowck reasons. I need both mutable and immutable access in this closure, and if there's a mutable borrow outside of the closure I can't get the immutable one inside.

Reply:
The underlying issue is that `get_satd` accepts `PlaneRegion` and `PlaneRegionMut` is not a `PlaneRegion`.